src/network/gossip.rs

```diff
@@ -374,13 +385,26 @@ impl SnapchainGossip {
         }
     }
 
+    pub fn update_slow_peers(&mut self, peer_id: &PeerId) {
+        let num_timeouts = self.slow_peers.get(peer_id).unwrap_or(0);
+        if num_timeouts <= 3 {
+            self.swarm
+                .behaviour_mut()
+                .gossipsub
+                .remove_blacklisted_peer(peer_id);
+        } else {
+            self.swarm.behaviour_mut().gossipsub.blacklist_peer(peer_id);
```

Comment:
I don't think blackslisting affects the connection (only rejects the messages), so this may not do what we want. I think we need to disconnect manually.